espy/output/zeek/dns.go

```diff
@@ -0,0 +1,221 @@
+		"RP":         "17",
+		"AFSDB":      "18",
+		"X25":        "19",
+		"ISDN":       "20",
+		"RT":         "21",
+		"NSAPPTR":    "23",
+		"SIG":        "24",
+		"KEY":        "25",
+		"PX":         "26",
+		"GPOS":       "27",
+		"AAAA":       "28",
+		"LOC":        "29",
+		"NXT":        "30",
+		"EID":        "31",
+		"NIMLOC":     "32",
+		"SRV":        "33",
+		"ATMA":       "34",
+		"NAPTR":      "35",
+		"KX":         "36",
+		"CERT":       "37",
+		"DNAME":      "39",
+		"OPT":        "41",
+		"APL":        "42",
+		"DS":         "43",
+		"SSHFP":      "44",
+		"RRSIG":      "46",
+		"NSEC":       "47",
+		"DNSKEY":     "48",
+		"DHCID":      "49",
+		"NSEC3":      "50",
+		"NSEC3PARAM": "51",
+		"TLSA":       "52",
+		"SMIMEA":     "53",
+		"HIP":        "55",
+		"NINFO":      "56",
+		"RKEY":       "57",
+		"TALINK":     "58",
+		"CDS":        "59",
+		"CDNSKEY":    "60",
+		"OPENPGPKEY": "61",
+		"CSYNC":      "62",
+		"ZONEMD":     "63",
+		"SVCB":       "64",
+		"HTTPS":      "65",
+		"SPF":        "99",
+		"UINFO":      "100",
+		"UID":        "101",
+		"GID":        "102",
+		"UNSPEC":     "103",
+		"NID":        "104",
+		"L32":        "105",
+		"L64":        "106",
+		"LP":         "107",
+		"EUI48":      "108",
+		"EUI64":      "109",
+		"URI":        "256",
+		"CAA":        "257",
+		"AVC":        "258",
+	}
+	qTypeID, ok := dnsMap[qType]
+	if !ok {
+		return "", fmt.Errorf("invalid DNS type: %s", qType)
+	}
+	return qTypeID, nil
+}
+
+type DnsTSV struct{}
+
+func (c DnsTSV) Header() TSVHeader {
+	return TSVHeader{
+		Separator:    "\\x09",
+		SetSeparator: ",",
+		EmptyField:   "(empty)",
+		UnsetField:   "-",
+		Path:         "dns",
+		Fields: []string{
+			"ts", "uid", "id.orig_h", "id.orig_p", "id.resp_h", "id.resp_p",
+			"proto", "trans_id", "rtt", "query", "qclass", "qclass_name",
+			"qtype", "qtype_name", "rcode", "rcode_name", "AA", "TC", "RD", "RA", "Z",
+			"answers", "TTLs", "rejected", "agent_hostname", "agent_uuid",
+		},
+		Types: []string{
+			"time", "string", "addr", "port", "addr", "port", "enum", "count",
+			"interval", "string", "count", "string", "count", "string", "count", "string",
+			"bool", "bool", "bool", "bool", "count", "vector[string]", "vector[interval]",
+			"bool", "string", "string",
+		},
+	}
+}
+
+func (c DnsTSV) FormatLines(outputData []input.ECSRecord) (output string, err error) {
+	var outputBuilder strings.Builder
+	header := c.Header()
+	//escape \\x09 to tab
+	separator, _ := strconv.Unquote(fmt.Sprintf("\"%s\"", header.Separator))
+
+	for i := range outputData {
+		goStartTime, err := time.Parse(time.RFC3339Nano, outputData[i].RFCTimestamp)
+		if err != nil {
+			return output, input.ErrMalformedECSRecord
+		}
+
+		answersSetBuilder := strings.Builder{}
+		answerTypeName := header.UnsetField
+		answerTypeID := header.UnsetField
+
+		// Don't report CNAME responses since Zeek doesn't
+		// WEIRD: Windows issues A queries for IP addresses and gets back "-" in the answers
```

Comment:
Example DNS query where this is an issue: [sysmon22weird.txt](https://github.com/activecm/espy/files/8166364/sysmon22weird.txt)